### DISCUSSION TITLE
Non-disclosure statements wanted

### DISCUSSION THREAD

**Discussion by jochen-schuettler** [sustainable-computing-io/kepler#1065] [General]
We have a lot of customers running their services inside Kubernetes clusters in cloud-based VMs. At least some of them are interested in measuring energy losses with kepler. But for trust reasons, we would like a positive statement that no data is transmitted to machines outside the monitored network. The documentation makes no statement to non-disclosure until now.

**Comment by marceloamaral** [comment]
Thanks @jochen-schuettler to opening this discussion!

Yes, Kepler only export the Prometheus metrics, no other data is transmitted outside the node.

@rootfs where should we include this description?

**Reply by rootfs** [reply]
@jochen-schuettler right,  kepler is a prometheus exporter, it doesn't transmit data outside the monitored host.

**Reply by rootfs** [reply]
@marceloamaral @jochen-schuettler would this statement sound good?

```When acting as a Prometheus exporter, Kepler daemonset doesn't transmit any data from the host to outside the cluster.```

**Reply by jochen-schuettler** [reply]
Sounds good, and additionally declare what's different when it doesn't act as Prometheus exporter? If this doesn't happen at all I'd rather:
`Kepler doesn't transmit any data from the host to outside the cluster.`

**Reply by rootfs** [reply]
@jochen-schuettler we are going to add open telemetry support to kepler. That means kepler can be configured by the end user to emit the telemetry data to a collector that is also controlled by the end user. The telemetry enablement needs to be controlled by end users, not by default.

For now, a prometheus exporter statement serves the purpose the best.
